File: src/main/java/phoenix/utils/entity/AbstractFlyingEntity.java
package phoenix.utils.entity;

import com.sun.javafx.geom.Vec3d;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.FlyingEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

public abstract class AbstractFlyingEntity  extends FlyingEntity
{
    public Vec3d orbitOffset = new Vec3d();
    public BlockPos orbitPosition = BlockPos.ZERO;
    public AttackPhases attackPhase = AttackPhases.CIRCLE;
    protected AbstractFlyingEntity(EntityType<? extends FlyingEntity> type, World worldIn) {
        super(type, worldIn);
    }
}

File: src/main/kotlin/phoenix/utils/entity/AbstractFlyingEntity.kt
package phoenix.utils.entity

import com.sun.javafx.geom.Vec3d
import net.minecraft.entity.EntityType
import net.minecraft.entity.FlyingEntity
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

abstract class AbstractFlyingEntity protected constructor(type: EntityType<out FlyingEntity>, worldIn: World) : FlyingEntity(type, worldIn)
{
    var orbitOffset = Vec3d()
    var orbitPosition = BlockPos.ZERO
    var attackPhase = AttackPhases.CIRCLE
}
